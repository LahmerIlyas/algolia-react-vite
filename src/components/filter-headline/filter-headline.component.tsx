/* eslint-disable react/prop-types */
import Typography from '../typography/typography.component'

interface FilterHeadlineProps {
  items: Array<{
    id: string | number
    label: string
    isSelected: boolean
  }>
  onClicked: (id: string) => void
  title: string
}

export const FilterHeadline: React.FC<FilterHeadlineProps> = (props) => {
  return (
    <div>
      <Typography as="h2" variant="s2" className="mb-4">{props.title}</Typography>
      {props.items.map(item => (
        <div key={item.id} className="flex flex-row items-center">
          <input
            className="mr-4"
            type="checkbox"
            value={item.label}
            checked={item.isSelected}
            onChange={() => props.onClicked(item.id)}
          />
          <Typography
            className="truncate"
            as="h5"
            variant="s5"
          >{item.label}
          </Typography>

        </div>
      ))}
    </div>
  )
}
