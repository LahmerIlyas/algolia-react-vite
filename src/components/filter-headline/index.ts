export * from './filter-headline.component'
