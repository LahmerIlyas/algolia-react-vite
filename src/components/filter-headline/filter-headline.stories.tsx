import React from 'react'
import type { ComponentMeta, ComponentStory } from '@storybook/react'

import { FilterHeadline } from './filter-headline.component'

export default {
  title: 'Example/FilterHeadline',
  component: FilterHeadline,
} as ComponentMeta<typeof FilterHeadline >

const Template: ComponentStory<typeof FilterHeadline> = () => (
  <React.Fragment>
    <FilterHeadline
      items={[
        { id: 1, isSelected: false, label: 'Test' },
        { id: 2, isSelected: false, label: 'Test 1' },
        { id: 3, isSelected: false, label: 'Test 2' },
        { id: 4, isSelected: false, label: 'Test 3' },
      ]}
      onClicked={() => { }}
      title="Categories"
    />
  </React.Fragment>
)

export const ExampleStory = Template.bind({})
ExampleStory.args = {
}
