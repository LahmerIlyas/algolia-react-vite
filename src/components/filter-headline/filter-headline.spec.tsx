import { describe, expect, it } from 'vitest'
import renderer from 'react-test-renderer'
import { FilterHeadline } from './filter-headline.component'

describe('Given filter headline element', () => {
  it('should renders correctly', () => {
    const items = [
      { id: 1, label: 'test', isSelected: false },
      { id: 2, label: 'test 2 ', isSelected: true },
    ]
    const tree = renderer.create(
      <FilterHeadline
        items={items}
        title={'Test title'}
        onClicked={() => {}}
      />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
