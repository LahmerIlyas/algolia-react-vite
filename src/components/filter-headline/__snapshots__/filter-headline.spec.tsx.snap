// Vitest Snapshot v1

exports[`Given filter headline element > should renders correctly 1`] = `
<div>
  <h2
    as="h2"
    className="text-s2 mb-4"
  >
    Test title
  </h2>
  <div
    className="flex flex-row items-center"
  >
    <input
      checked={false}
      className="mr-4"
      onChange={[Function]}
      type="checkbox"
      value="test"
    />
    <h5
      as="h5"
      className="text-s5 truncate"
    >
      test
    </h5>
  </div>
  <div
    className="flex flex-row items-center"
  >
    <input
      checked={true}
      className="mr-4"
      onChange={[Function]}
      type="checkbox"
      value="test 2 "
    />
    <h5
      as="h5"
      className="text-s5 truncate"
    >
      test 2 
    </h5>
  </div>
</div>
`;
